engine/api/repositoriesmanager/repogitlab/client.go

```diff
@@ -0,0 +1,373 @@
+		return gitlab.Pending
+	case sdk.StatusChecking:
+		return gitlab.Pending
+	case sdk.StatusBuilding:
+		return gitlab.Running
+	case sdk.StatusSuccess:
+		return gitlab.Success
+	case sdk.StatusFail:
+		return gitlab.Failed
+	case sdk.StatusDisabled:
+		return gitlab.Canceled
+	case sdk.StatusNeverBuilt:
+		return gitlab.Canceled
+	case sdk.StatusUnknown:
+		return gitlab.Failed
+	case sdk.StatusSkipped:
+		return gitlab.Canceled
+	}
+
+	return gitlab.Failed
+}
+
+//SetStatus set build status on Gitlab
+func (c *GitlabClient) SetStatus(event sdk.Event) error {
+	var eventpb sdk.EventPipelineBuild
+	if event.EventType != fmt.Sprintf("%T", sdk.EventPipelineBuild{}) {
+		return nil
+	}
+
+	if err := mapstructure.Decode(event.Payload, &eventpb); err != nil {
+		return err
+	}
+
+	log.Debug("Process event:%+v", event)
+
+	cdsProject := eventpb.ProjectKey
+	cdsApplication := eventpb.ApplicationName
+	cdsPipelineName := eventpb.PipelineName
+	cdsBuildNumber := eventpb.BuildNumber
+	cdsEnvironmentName := eventpb.EnvironmentName
+
+	key := fmt.Sprintf("%s-%s-%s",
+		cdsProject,
+		cdsApplication,
+		cdsPipelineName,
+	)
+
+	url := fmt.Sprintf("%s/project/%s/application/%s/pipeline/%s/build/%d?envName=%s",
+		uiURL,
+		cdsProject,
+		cdsApplication,
+		cdsPipelineName,
+		cdsBuildNumber,
+		url.QueryEscape(cdsEnvironmentName),
+	)
+
+	desc := fmt.Sprintf("Build #%d %s", eventpb.BuildNumber, key)
+
+	cds := "CDS"
+	opt := &gitlab.SetCommitStatusOptions{
+		Name:        &cds,
+		Context:     &cds,
+		State:       getGitlabStateFromStatus(eventpb.Status),
+		Ref:         &eventpb.BranchName,
+		TargetURL:   &url,
+		Description: &desc,
+	}
+
+	_, _, err := c.client.Commits.SetCommitStatus(eventpb.RepositoryFullname, eventpb.Hash, opt)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+//GetEvents is not implemented
+func (c *GitlabClient) GetEvents(repo string, dateRef time.Time) ([]interface{}, time.Duration, error) {
+	return nil, 0.0, fmt.Errorf("Not implemented on Gitlab")
+}
+
+//PushEvents is not implemented
+func (c *GitlabClient) PushEvents(string, []interface{}) ([]sdk.VCSPushEvent, error) {
+	return nil, fmt.Errorf("Not implemented on Gitlab")
+}
+
+//CreateEvents is not implemented
+func (c *GitlabClient) CreateEvents(string, []interface{}) ([]sdk.VCSCreateEvent, error) {
+	return nil, fmt.Errorf("Not implemented on Gitlab")
+}
+
+//DeleteEvents is not implemented
+func (c *GitlabClient) DeleteEvents(string, []interface{}) ([]sdk.VCSDeleteEvent, error) {
+	return nil, fmt.Errorf("Not implemented on Gitlab")
+}
+
+//PullRequestEvents is not implemented
+func (c *GitlabClient) PullRequestEvents(string, []interface{}) ([]sdk.VCSPullRequestEvent, error) {
+	return nil, fmt.Errorf("Not implemented on Gitlab")
+}
+
+// Release not implemented in Gitlab
+func (c *GitlabClient) Release(repo string, tagName string, title string, releaseNote string) (*sdk.VCSRelease, error) {
+	return nil, fmt.Errorf("Gitlab do not provide release system")
```

Comment:
https://docs.gitlab.com/ee/api/tags.html#create-a-new-release

Reply:
yeah I known haha, I'll change the error message, but I'll implement it later